main.cpp

```diff
@@ -31,14 +31,16 @@ void fnExit3(void) {
 int main(int argc, char *argv[]) {
     atexit(fnExit3);
 
-    if (argc == 1) {
-        std::cout << "Need one argument. Use argument 1 to start JasmineGraph in Master mode. Use any other integer to "
+    if (argc < 4) {
+        std::cout << "Need three arguments. Use argument '1 <serverPort> <serverDataPort>' to start JasmineGraph in Master mode. Use any other integer with <serverPort> <serverDataPort> to "
```

Comment:
We do not need to get <serverPort> and <serverDataPort> as arguments here. If we want to start as master we should be able to send 1 as the single integer parameter and get the service up and running. If we want it to start as worker it should also get worker ID. Using that the worker can resolve its server and data ports.

Reply:
I think the better way would be to use 2 as the worker mode and then if 2 is present we should expect another two arguments <serverPort> and <serverDataPort> to be present.